Backpack/src/main/java/net/skyscanner/backpack/calendar2/list/CalendarAdapter.kt

```diff
@@ -61,32 +53,22 @@ internal class CalendarAdapter(
 
   override fun getItemViewType(position: Int): Int = when (data[position]) {
     is CalendarCell.Day -> TYPE_DAY
-    is CalendarCell.Footer -> TYPE_FOOTER
     is CalendarCell.Header -> TYPE_HEADER
     is CalendarCell.Space -> TYPE_SPACE
   }
 
   override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder = when (viewType) {
     TYPE_HEADER -> CalendarCellHeaderHolder(parent)
     TYPE_DAY -> CalendarCellDayHolder(parent, output)
-    TYPE_FOOTER -> footerAdapter.onCreateViewHolder(parent)
     else -> CalendarCellSpaceHolder(parent)
   }
 
   override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) =
-    if (holder is ItemHolder<*>) {
-      holder as ItemHolder<CalendarCell>
-      holder.invoke(data[position])
-    } else {
-      val footers = footerAdapter as CalendarFooterAdapter<RecyclerView.ViewHolder>
-      val item = data[position] as CalendarCell.Footer
-      footers.onBindViewHolder(holder, item.yearMonth)
-    }
+    (holder as ItemHolder<CalendarCell>).invoke(data[position])
```

Comment:
You could update the viewholder generic in the constructor to avoid the casting here now that the footer adapter is gone